src/main/java/org/sagebionetworks/web/client/widget/entity/browse/EntityFinderViewImpl.java

```diff
@@ -109,6 +110,7 @@ public void onClick(ClickEvent event) {
 		cancelButton.addClickHandler(new ClickHandler() {
 			@Override
 			public void onClick(ClickEvent event) {
+				modal.setRemoveOnHide(true);
```

Comment:
If you open an entity finder, click cancel (and setRemoveOnHide=true so it is removed from the DOM), and then you pop the Entity Finder back up, does it still behave well?